packages/blade/src/components/Dropdown/docs/DropdownWithButton.stories.tsx

```diff
@@ -271,4 +284,53 @@ InternalLinkDropdown.parameters = {
   },
 };
 
+export const InternalIconButtonDropdown = (): React.ReactElement => {
+  const [status, setStatus] = React.useState<string | undefined>('latest-added');
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+
+  return (
+    <Box padding="spacing.10">
+      <Dropdown onOpenChange={setIsDropdownOpen} isOpen={isDropdownOpen}>
+        <DropdownIconButton
+          icon={BoxIcon}
+          tooltip={{ content: 'Check status' }}
+          accessibilityLabel="Status Dropdown"
+        />
```

Comment:
Do we need to expose a tooltip prop? 
It feels a bit off from an API perspective since I’d expect the component to support seamless composition and work directly with the Tooltip component - especially since it's a compound component. Something like:

```ts
<Dropdown>
  <Tooltip><DropdownButton /></Tooltip>
</Dropdown>
```

Reply:
+1 to Anurag.

Reply:
That is what I was thinking too but it breaks too many of Dropdown's assumptions because now Dropdown can't figure out what children it has.

One way is we go the nested route and find the eventual children of Tooltip but that feels too hacky and delicate so in this case I'm following what we followed in SideNavLink. There too we had similar issue and went with tooltip as a prop

Reply:
Discussed with Anurag. We can follow what we follow on Menu. So Tooltip can be wrapped around the Dropdown component itself with InteractiveWrapper-

```jsx
<Tooltip content="Check status">
  <TooltipInteractiveWrapper>
    <Dropdown>
      <DropdownIconButton />
      <DropdownOverlay>
      </DropdownOverlay>
    </Dropdown>
  </TooltipInteractiveWrapper>
</Tooltip>
```